### ISSUE TITLE
JsonToKotlinClass maven plugin

### ISSUE THREAD

**Issue by kezhenxu94** [wuseal/JsonToKotlinClass#147]
Now we generate the Kotlin data class and put it into the codebase, **other code generator (like Protobuf and [greendao](http://greenrobot.org/greendao/))** usually generate the code into a seperated folder (`target/generated-sources`) **automatically**, this would be really helpful when the JSON structure changes frequently, it won't pollute the commits history

![image](https://user-images.githubusercontent.com/15965696/56262606-086a4a80-6112-11e9-8100-f8731e3feb83.png)

**Comment by AlexTrotsenko** [comment]
@wuseal  any updates ? This would be lovely to see such kotlin plugin.
Check java plugin for example: https://github.com/joelittlejohn/jsonschema2pojo/tree/master/jsonschema2pojo-gradle-plugin

**Comment by theapache64** [comment]
Wow.. cool. @wuseal I'd vote for this feature :+1:

**Comment by hannesa2** [comment]
I'd vote for this feature 👍

**Comment by kassim** [comment]
Would be a great addition!

**Comment by eirnym** [comment]
It would be a  great addition!

**Comment by eirnym** [comment]
@wuseal I can write a plugin for you if you need it, it'd be a simple one :) 

Here you can see mine for jsonschema2pojo https://github.com/eirnym/js2p-gradle

**Comment by theapache64** [comment]
@eirnym @SealWu

**Comment by theapache64** [comment]
@eirnym That'd be cool. You can use our [library](https://github.com/wuseal/JsonToKotlinClass/blob/master/LIBRARY.md). 
What do you think @wuseal  ?

**Comment by eirnym** [comment]
Cool, @theapache64 I'd create one, how you'd like to name it?

**Comment by theapache64** [comment]
@wuseal makes the final call. let's wait for him. meantime, can you hope into our slack channel? Here's the [invite](https://join.slack.com/t/jsontokotlinclass/shared_invite/enQtNjk2NjM1MDM3MDA5LWYxYTI3ZDRlZTMyNDI2Mjg5NzZmZjQzM2ZlNjE5OTFmNWYyY2E0ZGEzZTQ3OWNhODEwOTUwYmU0YmRjZjQ5OGQ) link

**Comment by kezhenxu94** [comment]
As a plugin, I think we should create another repo to host it. Also, I recommend to create an organization and host these repos there for better collaboration and user recognition.

**Comment by theapache64** [comment]
@kezhenxu94 I agree

**Comment by hannesa2** [comment]
I agree https://github.com/eirnym/js2p-gradle/issues/8#issuecomment-774426610

**Comment by wuseal** [comment]
@eirnym Of course, That's awesome, then name can be `jsontokotlinclass` and id could be `wu.seal.jsontokotlin`. You could create a dir named `plugin` in the root repo.
 Welcome to open a PR here, and I and @kezhenxu94 will review it

**Comment by huehnerlady** [comment]
Is there any update on this? sounds like exactly what I need, but I cannot find such a plugin so far 🤔
